php/class-wp-cli.php

```diff
@@ -21,11 +48,77 @@ public function create_guest_authors( $args, $assoc_args ) {
 		global $coauthors_plus;
 
 		$defaults = array(
-				// There are no arguments at this time
-			);
+			'roles' => array( 'administrator', 'editor', 'author', 'contributor' ),
+			'batch-size' => 1000,
+		);
 		$this->args = wp_parse_args( $assoc_args, $defaults );
 
-		$users = get_users();
+		if ( is_array( $this->args['roles'] ) ) {
+			$role_whitelist = $this->args['roles'];
+		} else {
+			$role_whitelist = explode( ',', $this->args['roles'] );
+		}
+
+		if ( $this->args['batch-size'] > 0 && is_numeric( $this->args['batch-size'] ) ) {
+			$batch_size = (int) $this->args['batch-size'];
+		} else {
+			WP_CLI::error( __( 'Batch size must be an integer greater than zero.', 'co-authors-plus' ) );
+		}
+
+		foreach ( $role_whitelist as $role ) {
+			// Does this role exist in this instance?
+			if ( ! $GLOBALS['wp_roles']->is_role( $role ) ) {
+				WP_CLI::error( __( 'Role ', 'co-authors-plus' ) . $role . __( ' does not exist.', 'co-authors-plus' ) );
```

Comment:
Should we instead inject $role as a variable inside a singular translation? That would seem to be best for many languages where the sentence structure may be different.

Reply:
:+1: